File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys

input = sys.stdin.readline

n,m = map(int,input().split())
lst = list(map(int,input().split()))

s=0
e=1
ans = 0
while s<=e and e<=n:
    sm = 0
    for i in range(s,e):
        sm += lst[i]
    if sm > m:
        s += 1
    elif sm < m:
        e += 1
    else:
        e += 1
        ans += 1

print(ans)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.js
let input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const [N, M] = input[0].split(" ").map(Number);
const lst = input[1].split(" ").map(Number);

let s = 0;
let e = 1;
let cnt = 0;

while (e <= N && s <= e) {
  let sm = 0;
  for (let i = s; i < e; i++) {
    sm += lst[i];
  }
  if (sm < M) {
    e += 1;
  } else if (sm > M) {
    s += 1;
  } else {
    cnt += 1;
    e += 1;
  }
}
console.log(cnt);